server/data/nomisClientBuilder.ts

```diff
@@ -0,0 +1,214 @@
+    agent: agentOptions,
+    timeout: timeoutSpec,
+  })
+  const oauthRestClient = buildRestClient(tokenSource, authUrl, 'OAuth API', {
+    agent: agentOptions,
+    timeout: timeoutSpec,
+  })
+
+  const addReleaseDatesToPrisoner = pipe(
+    addReleaseDate,
+    addEffectiveConditionalReleaseDate,
+    addEffectiveAutomaticReleaseDate
+  )
+
+  return {
+    getBooking(bookingId) {
+      return nomisRestClient.getResource(`/bookings/${bookingId}`)
+    },
+
+    getBookingByOffenderNumber(offenderNo) {
+      return nomisRestClient.getResource(`/bookings/offenderNo/${offenderNo}`)
+    },
+
+    getAliases(bookingId) {
+      return nomisRestClient.getResource(`/bookings/${bookingId}/aliases`)
+    },
+
+    getIdentifiers(bookingId) {
+      return nomisRestClient.getResource(`/bookings/${bookingId}/identifiers`)
+    },
+
+    getMainOffence(bookingId) {
+      return nomisRestClient.getResource(`/bookings/${bookingId}/mainOffence`)
+    },
+
+    getImageInfo(imageId) {
+      return nomisRestClient.getResource(`/images/${imageId}`)
+    },
+
+    async getHdcEligiblePrisoners() {
+      const path = `/offender-sentences/home-detention-curfew-candidates`
+      const headers = { 'Page-Limit': 10000 }
+
+      const prisoners = await nomisRestClient.getResource(path, headers)
+      return prisoners.map(addReleaseDatesToPrisoner)
+    },
+
+    async getOffenderSentencesByNomisId(nomisIds, batchSize = 50) {
+      const path = `/offender-sentences`
+      if (isEmpty(nomisIds)) {
+        return []
+      }
+
+      const prisoners = await batchRequests(nomisIds, batchSize, (batch) => {
+        const query = { offenderNo: batch }
+        return nomisRestClient.getResource(path, { 'Page-Limit': batchSize }, query)
+      })
+
+      return prisoners.map(addReleaseDatesToPrisoner)
+    },
+
+    async getOffenderSentencesByBookingId(bookingIds, addReleaseDates = true) {
+      const path = `/offender-sentences/bookings`
+      const headers = { 'Page-Limit': 10000 }
+      const body = [].concat(bookingIds)
+
+      const prisoners = await nomisRestClient.postResource(path, body, headers)
+
+      if (!addReleaseDates) {
+        return prisoners
+      }
+
+      return prisoners.map(addReleaseDatesToPrisoner)
+    },
+
+    async getImageData(id) {
+      const image = await nomisRestClient.getResource(`/images/${id}/data`)
+      if (image) {
+        return image
+      }
+      throw Error('Not Found')
+    },
+
+    getEstablishment(agencyLocationId) {
+      return nomisRestClient.getResource(`/agencies/prison/${agencyLocationId}`)
+    },
+
+    getUserInfo(userName) {
+      return oauthRestClient.getResource(`/api/user/${userName}`)
+    },
+
+    getLoggedInUserInfo(): Promise<Profile> {
+      return oauthRestClient.getResource(`/api/user/me`)
+    },
+
+    getUserRoles(): Promise<Role> {
+      return oauthRestClient.getResource(`/api/user/me/roles`)
```

Comment:
Out of scope for this story but is there anything to stop us from splitting out the auth calls into a separate client?